packages/analytics-core/src/plugins/destination.ts

```diff
@@ -59,7 +59,7 @@ export class Destination implements DestinationPlugin {
 
     this.storageKey = `${STORAGE_PREFIX}_${this.config.apiKey.substring(0, 10)}`;
     const unsent = await this.config.storageProvider?.get(this.storageKey);
-    this.saveEvents(); // sets storage to '[]'
+    await this.saveEvents(); // sets storage to '[]'
```

Comment:
all these `await`s have performance impact. please use cautiously

Reply:
As all events should be in queue, I deleted `get()` so all await should not be needed.